File: solution/day20/src/test/java/builders/DiceBuilder.java
package builders;

import java.util.Arrays;

public class DiceBuilder {
    private final int[] dice;

    private DiceBuilder(int[] dice) {
        this.dice = dice;
    }

    public static DiceBuilder newRoll(int dice1, int dice2, int dice3, int dice4, int dice5) {
        return new DiceBuilder(new int[]{dice1, dice2, dice3, dice4, dice5});
    }

    public int[] build() {
        return dice;
    }

    @Override
    public String toString() {
        return Arrays.toString(dice);
    }
}

File: exercise/kotlin/day20/src/test/kotlin/DiceBuilder.kt
class DiceBuilder private constructor(private val dice: IntArray) {
    fun build(): IntArray {
        return dice
    }

    override fun toString(): String = dice.contentToString()

    companion object {
        fun `🎲🎲🎲🎲🎲`(dice1: Int, dice2: Int, dice3: Int, dice4: Int, dice5: Int): DiceBuilder =
            DiceBuilder(intArrayOf(dice1, dice2, dice3, dice4, dice5))
    }
}